homeassistant/components/hdmi_cec/__init__.py

```diff
@@ -214,6 +218,18 @@ def setup(hass: HomeAssistant, base_config):
         adapter = CecAdapter(name=display_name[:12], activate_source=False)
     hdmi_network = HDMINetwork(adapter, loop=loop)
 
+    def _adapter_watchdog(now=None):
+        _LOGGER.debug("Reached _adapter_watchdog")
+        event.async_call_later(hass, WATCHDOG_INTERVAL, _adapter_watchdog)
```

Comment:
There is no way to cancel this, you should put some logic to handle this only when disconnected. You can use async_track_point_in_utc_time to store a handle that you can cancel once you get reconnected.

Reply:
We don't need to cancel this. The watchdog function is what checks to see if we are disconnected or not. So it must run even when we are connected. Otherwise when disconnected, we will just silently fail to update states.

This stems from the fact that the pycec library pushes state changes (we don't poll it), but doesn't push a state change on disconnect. So we have to actively check if it is connected. I don't think this is a problem because checking doesn't do any i/o, it just checks a property of the pycec adapter class.

Reply:
Ok